### PR TITLE
Improve level of detail in error reporting.

### PR DESCRIPTION
**Pull request by bemasc**
This is motivated by https://github.com/uProxy/uproxy/issues/800

### CODE REVIEW

File: providers/core.tcpsocket.js [hunk 1]
```diff
@@ -262,12 +286,25 @@ Socket_chrome.errorStringOfCode = function(code) {
       'UNKNOWN';
 };
 
+/**
+ * Get the Freedom error code associated with a Chrome error string
+ * @method freedomErrorCode
+ * @static
+ * @private
+ * @param {string} errorString The Chrome error string
```
Comment (willscott) [inline providers/core.tcpsocket.js]: this is presumably `String=` in the same way that it is `Number=` in errorStringOfCode?

File: providers/core.tcpsocket.js [hunk 2]
```diff
@@ -262,12 +286,25 @@ Socket_chrome.errorStringOfCode = function(code) {
...
+};
+
 /*
  * Alert a consumer that a socket is disconnected, with appropriate
  * error message.
  * @method dispatchDisconnect
  * @private
- * @param {Number} code the code returned by chrome when the socket closed.
+ * @param {Number=} code the code returned by chrome when the socket closed.
+ * @return {!Object} Error JSON object.
```
Comment (willscott) [inline providers/core.tcpsocket.js]: could be documented better. Either
closure style record type: https://developers.google.com/closure/compiler/docs/js-for-compiler#records
or yui style: http://stackoverflow.com/questions/15396747/javascript-yuidoc-how-to-document-an-object-parameter-of-a-method/24795635#24795635

File: providers/core.tcpsocket.js (hunk 1 above)
Comment (bemasc) [inline providers/core.tcpsocket.js]: Done.

File: providers/core.tcpsocket.js (hunk 2 above)
Comment (bemasc) [inline providers/core.tcpsocket.js]: Done.

File: providers/core.tcpsocket.js [hunk 3]
```diff
@@ -262,38 +286,52 @@ Socket_chrome.errorStringOfCode = function(code) {
...
+  } else if(freedomErrorCode === 'CONNECTION_CLOSED') {
+    errorMessage = 'Connection closed gracefully';
+  } else {
+    errorMessage = 'Socket Error: ' + code + ': ' + errorString;
   }
 
-  Socket_chrome.removeActive(this.id);
-  delete this.id;
+  var errorObject = {
+    'errorcode': freedomErrorCode,
```
Comment (trevj) [inline providers/core.tcpsocket.js]: Typo?

It's preventing the `errcode` field from getting through:
https://github.com/freedomjs/freedom-for-chrome/pull/62

**Pull request merged**